File: src/engine/commands/whocommand.cpp
#include "whocommand.h"

#include "realm.h"


#define super Command

WhoCommand::WhoCommand(QObject *parent) :
    super(parent) {

    setDescription("Show who is currently online.\n"
                   "\n"
                   "Example: who");
}

WhoCommand::~WhoCommand() {
}

void WhoCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    GameObjectPtrList players = realm()->onlinePlayers();
    if (players.length() == 1) {
        send("Only you are online.");
    } else {
        send("%1 are online.", players.joinFancy(Capitalized));
    }
}

File: data/commands/whocommand.js
function WhoCommand() {
    Command.call(this);

    this.setDescription("Show who is currently online.\n" +
                        "\n" +
                        "Example: who");
}

WhoCommand.prototype = new Command();
WhoCommand.prototype.constructor = WhoCommand;

WhoCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    var players = Realm.onlinePlayers();
    if (players.length === 1) {
        this.send("Only you are online.");
    } else {
        this.send("%1 are online.", players.joinFancy(Options.Capitalized));
    }
};

CommandRegistry.registerCommand("who", new WhoCommand());
